examples/pytorch/infograph/qm9_v2.py

```diff
@@ -0,0 +1,213 @@
+import numpy as np
+import os
+from tqdm import tqdm
+
+import torch as th
+
+import dgl
+from dgl.data.dgl_dataset import DGLDataset
+from dgl.data.utils import download, load_graphs, _get_dgl_url, extract_archive
+
+class QM9Dataset_v2(DGLDataset):
+    r"""QM9 dataset for graph property prediction (regression)
+    This dataset consists of 13,0831 molecules with 19 regression targets.
+    Node means atom and edge means bond.
```

Comment:
Nodes correspond to atoms and edges correspond to bonds. Is this true for QM9? I thought the edges are distance-based.

Reply:
There are two versions of QM9. In this version, edges are definite bonds between atoms. We can even assign each graph with its corresponding molecular structure.

This class is just for this example. I recommend you review this PR #2704. After the new QM9EdgeDataset is merged, I will update this example using the QM9EdgeDataset.

Reply:
But adding distance-based edge is still a common practice in modeling molecular graphs(as they are really sparse). In this case, we will add another dimension of the edge feature indicting whether this edge indeed exists(1 means yes while 0 means no).

Reply:
Sounds good.